packages/react-devtools-shared/src/devtools/views/useResizeObserver.js

```diff
@@ -0,0 +1,97 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ */
+
+import {useEffect, useState} from 'react';
+import ResizeObserver from 'resize-observer-polyfill';
+
+type SizeState = {|
+  height: number,
+  width: number,
+|};
+
+type RefSetterFunction = (current: HTMLElement | null) => void;
+
+function safeStateUpdater(setSizeState, width, height) {
+  setSizeState(prevState => {
+    if (prevState.height === height && prevState.width === width) {
+      return prevState;
+    } else {
+      return {height, width};
+    }
+  });
+}
+
+export default function useResizeObserver(): {|
+  ref: RefSetterFunction,
+  height: number,
+  width: number,
+|} {
+  const [sizeState, setSizeState] = useState<SizeState>({
+    height: 0,
+    width: 0,
+  });
+
+  const [element, setRef] = useState<HTMLElement | null>(null);
+
+  useEffect(() => {
+    if (element != null) {
+      const rect = element.getBoundingClientRect();
+      safeStateUpdater(setSizeState, rect.width, rect.height);
+
+      const resizeObserver = new ResizeObserver(entries => {
+        if (!Array.isArray(entries) || entries.length === 0) {
```

Comment:
This callback never fires when running inside a Firefox extension. It works fine in Chrome/Edge as well as Firefox inline.